### ISSUE TITLE
Doesn't work with Storybook v5

### ISSUE THREAD

**Issue by kristof0425** [tuchk4/storybook-readme#111]
I know it's super early as it's in alpha, but we really depend on this package and it's not compatible with the latest alpha releases.

Do you have plans to make this package support the new Storybook versions?

Thanks in advance!

**Comment by ndelangen** [comment]
I haven't tried it, so I don't know what the problem is. any info you can share would help.

There's an addon in the V5 core that supports markdown too btw.

**Comment by tuchk4** [comment]
@kristof0425 yes, I always try to support all new storybook versions and even alphas as soon as possible :)

@ndelangen could you pleasse give the link with docs about core markdon support?

**Comment by kristof0425** [comment]
@ndelangen I forgot to add the error description.

The problem is with `withReadme` hoc. It doesn't render anything in its panel.

This is my Storybook config:
```js
import React from 'react'
import path from 'path'

import { ThemeDecorator } from './decorators/index'
import {
  configure,
  storiesOf,
  getStorybook,
  addDecorator,
} from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import { withOptions } from '@storybook/addon-options'

const getPackageName = filePath =>
  path
    .dirname(filePath)
    .split(path.sep)
    .reverse()[1]

const req = require.context(
  '../../../components',
  true,
  /^((?!node_modules).)*\.example\.js$/
)

const readMeReq = require.context(
  '../../../components',
  true,
  /^((?!node_modules).)*\README\.md$/
)

addDecorator(
  withOptions({
    name: 'MK-Components',
  })
)

configure(() => {
  req.keys().forEach(pathToExample => {
    const { name, Example, options = {} } = req(pathToExample)
    const packageName = getPackageName(pathToExample)
    const readmePath = readMeReq.keys().find(rm => rm.includes(packageName))
    const readme = readMeReq(readmePath)
    console.log({readme})
    storiesOf(packageName, module)
      .addDecorator(withKnobs)
      .addDecorator(withReadme(readme))
      .addDecorator(withOptions(options))
      .addDecorator(ThemeDecorator)
      .add(name, () => <Example />, { options })
  })
}, module)

export { getStorybook }
```

You can see that I'm logging out the readmes and they are printed to the console just fine. Though the readme addon's panel in the DOM is empty.

This is what the console says, however I haven't found anything useful in it:

<img width="1440" alt="screenshot 2019-01-19 at 9 53 17" src="https://user-images.githubusercontent.com/20532589/51424653-1838c800-1bd1-11e9-8c88-b0ce3e85dbb4.png">

cc @tuchk4 

Let me know if I can help you in any other way!

**Comment by Rolandisimo** [comment]
Still very much a needed change. We also depend on READMEs. I will maybe have time to look into this on the weekend.

**Comment by enwin** [comment]
Hi there!

Now that storybook v5 is in RC I decided to give it a go and I'm facing the same issue. Using `.addDecorator(withReadme(readme))` gives me a blank panel.

I dug a bit in the code and compared it to `@storybook/addon-knobs` to have an idea. I found that the `onStory` property of the `api` object passed when registering the component is noted as [deprecated](https://github.com/storybooks/storybook/blob/99223b147fb19f46b341dc7627a4ed13f6f5c561/lib/ui/src/core/channel.js#L24).

Comparing to `addon-knobs`, the `api` object is passed instead of just the `onStory` property.

[What the component does now](https://github.com/tuchk4/storybook-readme/blob/b1a9a2dbbd4cfbcc0351b97f91c7b0dbf1332251/packages/storybook-readme/src/register.js#L19)

[What `addon-knobs` is doing](https://github.com/storybooks/storybook/blob/34ef10f78c9a39e102dca88fb43daf9d5fd9f63a/addons/knobs/src/register.js#L11)

On `componentDidMount` they are registering the events [differently](https://github.com/storybooks/storybook/blob/34ef10f78c9a39e102dca88fb43daf9d5fd9f63a/addons/knobs/src/components/Panel.js#L37)

The issue that I found with reacting to `onStory` is that the `kind` value passed is not consistent. On mount the value is the name of the story defined in your story file when using `storiesOf('My Story', …)` but on change the value is the formated id. `My Story` will become `my-story`.

The match cannot be done in `docManager.js` hence no readme is found to be displayed.

I hope this will help you fixing the issue. I cannot do it myself unfortunately, I'm not proficient enough in react to do that 😅

**Comment by Rolandisimo** [comment]
@enwin thanks for the digging. Definitely helps.

**Comment by EnjoyChan** [comment]
It doesn't render anything too.

**Comment by tuchk4** [comment]
everything will work after releasing #119

Right now released alpha `storybook-readme@5.0.0-alpha.150`

Note that right now there are __breaking changes at alpha version__. I wrote about here https://github.com/tuchk4/storybook-readme/issues/118#issuecomment-472684902

**Comment by tuchk4** [comment]
Close this issue 

Any discussions here https://github.com/tuchk4/storybook-readme/pull/119
